lib/services/sshAdapter.js

```diff
@@ -0,0 +1,58 @@
+"use strict";
+
+var Bluebird = require("bluebird");
+
+function SshAdapter (ec2) {
+	var SSH_POLLING_INTERVAL = 30; /*Time in Seconds*/
+
+	this.checkIfSshable = function (instanceID) {
+		var self = this;
+		return this.canSsh(instanceID)
+		.then(function (sshable) {
+			if (sshable) {
+				return sshable;
+			}
+			else {
+				return Bluebird.delay(SSH_POLLING_INTERVAL * 1000)
+				.then(self.checkIfSshable.bind(self, instanceID));
+			}
+		}.bind(self));
+	};
+
+	this.startSshPolling = function (instanceID) {
```

Comment:
This looks like it does exactly the same thing as `checkIfSshable()`. Do we really need both?